Sources/_SwiftSyntaxTestSupport/SourceEditsTestUtils.swift

```diff
@@ -0,0 +1,99 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+import XCTest
+
+/// Get `ConcurrentEdits` in source whose edited zones are marked with markers
+/// Also extract the markers from source to get original source and edited source
+public func getConcurrentEditsAndSources(_ source: String) -> (edits: ConcurrentEdits, orignialSource: String, editedSource: String) {
+  var editedSource = source
+
+  var sequentialEdits: [SourceEdit] = []
+  while let (edit, afterSource) = nextMarkedEdits(editedSource) {
+    sequentialEdits.append(edit)
+    editedSource = afterSource
+  }
+
+  return (ConcurrentEdits(fromSequential: sequentialEdits), extractOriginalSource(from: source), editedSource)
+}
+
+fileprivate func nextMarkedEdits(_ source: String) -> (edit: SourceEdit, editedSource: String)? {
+  guard let startIndex = source.firstIndex(where: { $0.isStartMarker }),
+    let separateIndex = source.firstIndex(where: { $0.isSeperateMarker }),
+    let endIndex = source.firstIndex(where: { $0.isEndMarker })
```

Comment:
Technically, I think you want to find the first separate marker after the start marker, and end marker after the separate marker. Otherwise, you can get into weird issues of ranges where the end location is before the start location when someone uses an incorrect format string like `⏩️class⏪️struct⏸️ foo {`.

```suggestion
    let separateIndex = source[startIndex...].firstIndex(where: { $0.isSeperateMarker }),
    let endIndex = source[separateIndex...].firstIndex(where: { $0.isEndMarker })
```